File: 11.Greedy/nonOverlap.2.py
def nonOverlapIntervals(self, A):
    if not A or len(A) == 0:
        return 0

    A.sort(key=lambda x: x[0])

    preStart = float('inf')
    ans = 0

    N = len(A)

    for i in range(N-1,-1,-1):
        start = A[i][0]
        end = A[i][1]

        if end <= preStart:
            ans += 1
            preStart = start

    return ans

File: 11.Greedy/nonOverlap.2.cpp
int nonOverlapIntervals(vector<vector<int>> &A) {
  const int N = A.size();

  sort(A.begin(), A.end());

  int preStart = INT_MAX;
  int ans = 0;
  for (int i = N - 1; i >= 0; i--) {
    const int end = A[i][1];
    const int start = A[i][0];
    if (end <= preStart) {
      ans++;
      preStart = A[i][0];
    }
  }

  return ans;
}
